src/query.ts

```diff
@@ -0,0 +1,44 @@
+// export type JSONValue = string | boolean | number
+
+export interface JSONValue {
+  string?: string
+  bool?: boolean
+  float?: number
+}
+
+export enum JSONOperation {
+  Eq = 0,
+  Ne,
+  Gt,
+  Lt,
+  Ge,
+  Le,
+}
+
+export interface JSONCriterion {
+  fieldPath?: string
+  operation?: JSONOperation
+  value?: JSONValue
+  query?: JSONQuery
+}
+
+export interface JSONSort {
+  fieldPath: string
+  desc: boolean
+}
+
+export interface JSONQuery {
```

Comment:
These are the minimal set of interfaces etc required to fulfil the JSONQuery interface. We should certainly add a QueryBuilder class/interface to make this easier to use.

Reply:
Sweet!